semgrep_output_v1.atd

```diff
@@ -128,7 +128,7 @@ type core_match_extra
    * finding. *)
   ?rendered_fix: string option;
 
-  engine_kind: engine_kind;
+  ?engine_kind: engine_kind option;
```

Comment:
Here it's fine, you don't need the ?, because it's the interface between semgrep-core and semgrep CLI.
It's not public, nobody external should rely on it, so we can do whatever we want.

Reply:
In that case, could we make this just `engine_kind: engine_kind`? I believe that `semgrep-core` should only be able to inject into the `Some` case here, by construction.

Reply:
yes, you can do engine: engine_kind.